aptos-move/block-executor/src/view.rs

```diff
@@ -1429,11 +1429,28 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>> LatestView<'a, T, S> {
         layout: UnknownOrLayout,
         kind: ReadKind,
     ) -> PartialVMResult<ReadResult> {
-        debug_assert!(
-            !state_key.is_module_path(),
-            "Reading a module {:?} using ResourceView",
-            state_key,
-        );
+        let msg = if state_key.is_module_path() {
+            Some(format!(
+                "Reading a module {:?} using ResourceView",
+                state_key
+            ))
+        } else if state_key.is_resource_group_path() && kind != ReadKind::Metadata {
```

Comment:
So for groups, we store metadata as a dummy resource right here? that's why metadata check?